CONTRIBUTING.md

```diff
@@ -20,42 +20,40 @@ file.
 From a separate terminal, start running a flutter app to connect to DevTools:
 - `git clone https://github.com/flutter/gallery.git` (this is an existing application with many examples of Flutter widgets)
 - `cd gallery`
+- ensure your flutter channel is the one required by the [gallery documentation](https://github.com/flutter/gallery#running-flutter-gallery-on-flutters-master-channel)
 - ensure the iOS Simulator is open (or a physical device is connected)
 - `flutter run`
-
-## Workflow for making changes
-
-- Create a branch from your cloned repo: `git checkout -b myBranch`
-- Commit work to your branch: `git commit -m “description”`
-- Push to your branch: `git push origin myBranch`
-- Navigate to the Pull Requests tab in the main [DevTools repo](https://github.com/flutter/devtools). You should see a popup to create a pull request from your local branch to DevTools master. Create a pull request.
-
-### Keeping your fork in-sync
-
-- Fetch branches/commits from the upstream DevTools: `git fetch upstream`
-- From your local branch, merge in the upstream master branch: `git merge upstream/master`
+- copy the "Observatory debugger and profiler" uri printed in the command output, to connect to the app from DevTools later
 
 ## Development
 
-To run DevTools as a Flutter web app, follow these steps.
-
-First, perform one-time setup:
-
-- `flutter config --enable-web`
-
-Now you can run the app at any time with the flutter command.
-
-From the packages/devtools_app directory:
+To run DevTools as a Flutter web app, from the packages/devtools_app directory:
 
 - `flutter run -d chrome`
 
+*NOTE:* Though DevTools is shipped as a Flutter Web app, we recommend developing as a Flutter Desktop app where possible for a more efficient development workflow. Please see the [Desktop Embedder] section below for instructions on running DevTools as a Flutter Desktop app.
+
 To test release performance:
 
 - `flutter run -d web-server --release --dart-define=FLUTTER_WEB_USE_SKIA=true`
 
 You can also use `-d headless-server`, which will start a headless server that serves the HTML
 files for the DevTools Flutter app.
 
+To connect to your running application, paste the earlier copied observatory URL into the section "Connect to a Running App" in DevTools.
+
+## Workflow for making changes
```

Comment:
sorry for the churn here, but actually on second thought, let's move the "## Workflow for making changes" and "### Keeping your fork in-sync" sections to _before_ "## Development prep". I didn't realize before just how long the "Development" section and subsections are.

Reply:
done